### PR TITLE
feat: reintroduce clone and rewind for cursors

### PR DESCRIPTION
**Pull request by mbroadst**
## Description
These two commits add back support for cloning and rewinding cursors.

**What changed?**
The changes should be fairly straightforward, but I wanted to call out a few design decisions:
~- I had originally intended `clone` to be an abstract method on `AbstractCursor`, forcing us to implement it for all possible cursor subclasses. This would have unfortunately required us to make `AbstractCursor` generic over the subclass types, since the following signature results in type ambiguity at the point of implementation in the subclass: `abstract clone(): this`~

- Therefore, `clone` is implemented by hand in `AbstractCursor` subclasses. I only implemented it for `FindCursor` and `AggregateCursor` because it's likely that's where legacy users have actually used this feature.

- There is a correction in the `rewind` implementation regarding sessions. When a session is started as part of cursor initialization I had originally marked this as an `explicit: true` start, however, this is clearly an implicit session the driver is creating in the absence of an explicitly provided one. 

**Are there any files to ignore?**
none

### CODE REVIEW

File: src/cursor/find_cursor.ts [hunk 1]
```diff
@@ -49,6 +49,16 @@ export class FindCursor extends AbstractCursor {
     }
   }
 
+  /**
+   * Returns a new unitialized copy of this cursor, with options matching those that have been set on the current instance
```
Comment (emadum) [inline src/cursor/find_cursor.ts:53]:
```suggestion
   * Returns a new uninitialized copy of this cursor, with options matching those that have been set on the current instance
```

**Review by emadum — APPROVED** [review]
LGTM, mod one typo

File: src/cursor/find_cursor.ts (hunk 1 above)
Comment (mbroadst) [reply src/cursor/find_cursor.ts]: fixed here and in aggregation_cursor too!

File: src/cursor/find_cursor.ts [hunk 2]
```diff
@@ -49,6 +49,16 @@ export class FindCursor extends AbstractCursor {
...
 
+  /**
+   * Returns a new uninitialized copy of this cursor, with options matching those that have been set on the current instance
+   */
+  clone(): FindCursor {
+    return new FindCursor(this.topology, this.namespace, this[kFilter], {
+      ...this[kBuiltOptions],
+      ...this.cursorOptions
+    });
+  }
```
Comment (nbbeeken) [inline src/cursor/find_cursor.ts:60]: I think you can still make this an abstract contract, the method doesn't return `this` it returns a new instance so the AbstractCursor would have:
```ts
abstract clone(): AbstractCursor;
```
and FindCursor etc. can have the more specific subclass return type.

**Review by nbbeeken — APPROVED** [review]
LGTM one suggestion
If we do want the abstract clone I think it's still available to us:

File: src/cursor/find_cursor.ts (hunk 2 above)
Comment (mbroadst) [reply src/cursor/find_cursor.ts:57]: duh, I should have thought of that :) implemented

**Pull request merged**